### PR TITLE
[Core] Add liquidation field in Trade

### PR DESCRIPTION
**Pull request by makarid**

### CODE REVIEW

File: xchange-core/src/main/java/org/knowm/xchange/dto/marketdata/Trade.java [hunk 1]
```diff
@@ -130,6 +134,10 @@ public String getTakerOrderId() {
     return takerOrderId;
   }
 
+  public boolean liquidation() {
```
Comment (walec51) [inline xchange-core/src/main/java/org/knowm/xchange/dto/marketdata/Trade.java:137]: we use the `get...` convention

File: xchange-btcmarkets/src/main/java/org/knowm/xchange/btcmarkets/BTCMarketsAdapters.java [hunk 2]
```diff
@@ -277,7 +277,8 @@ public static Trades adaptMarketTrades(
               marketTrade.getTimestamp(),
               marketTrade.getId().toString(),
               null,
-              null);
+              null,
+                  false);
```
Comment (walec51) [inline xchange-btcmarkets/src/main/java/org/knowm/xchange/btcmarkets/BTCMarketsAdapters.java:281]: formatting seams off in many places

File: xchange-core/src/main/java/org/knowm/xchange/dto/trade/UserTrade.java [hunk 3]
```diff
@@ -56,9 +56,10 @@ public UserTrade(
       String orderId,
       BigDecimal feeAmount,
       Currency feeCurrency,
-      String orderUserReference) {
+      String orderUserReference,
+      boolean liquidation) {
 
-    super(type, originalAmount, instrument, price, timestamp, id, null, null);
+    super(type, originalAmount, instrument, price, timestamp, id, null, null, liquidation);
```
Comment (walec51) [inline xchange-core/src/main/java/org/knowm/xchange/dto/trade/UserTrade.java:62]: isn't `liquidation` the type of information that would only make sense to show in a `UserTrade`?

File: xchange-core/src/main/java/org/knowm/xchange/dto/marketdata/Trade.java [hunk 4]
```diff
@@ -53,16 +54,18 @@ public class Trade implements Serializable {
    * @param id The id of the trade
    * @param makerOrderId The orderId of the maker in the trade
    * @param takerOrderId The orderId of the taker in the trade
+   * @param liquidation true if the order is a liquidation order
```
Comment (walec51) [inline xchange-core/src/main/java/org/knowm/xchange/dto/marketdata/Trade.java:57]: how many exchanges offer this type of information in their trades?

**Comment by makarid** [comment]
> @makarid are these PRs related? #4622

I don't think so

File: xchange-btcmarkets/src/main/java/org/knowm/xchange/btcmarkets/BTCMarketsAdapters.java (hunk 2 above)
Comment (makarid) [reply xchange-btcmarkets/src/main/java/org/knowm/xchange/btcmarkets/BTCMarketsAdapters.java:281]: the was an error on the auto-formating on my IDE i will fix it

File: xchange-core/src/main/java/org/knowm/xchange/dto/trade/UserTrade.java (hunk 3 above)
Comment (makarid) [reply xchange-core/src/main/java/org/knowm/xchange/dto/trade/UserTrade.java:62]: No, in some exchanges it is a public information for every trade.

File: xchange-core/src/main/java/org/knowm/xchange/dto/marketdata/Trade.java (hunk 4 above)
Comment (makarid) [reply xchange-core/src/main/java/org/knowm/xchange/dto/marketdata/Trade.java:57]: Let me check

**Comment by walec51** [comment]
@makarid please push review fixes if you would like this to get merged

**Comment by walec51** [comment]
is there any progress on this PR?

if not I will have to close it

**Pull request closed without merging**